integration_tests/src/main/python/asserts.py

```diff
@@ -26,23 +26,26 @@
 import difflib
 import sys
 
-def _assert_equal(cpu, gpu, float_check, path):
+def _assert_equal(cpu, gpu, float_check, path, cpu_data_adapted_from = "CPU"):
+    """
+    :param cpu_data_adapted_from: indicate where the cpu result is adapted from, e.g.: pyarrow
+    """
     t = type(cpu)
     if (t is Row):
-        assert len(cpu) == len(gpu), "CPU and GPU row have different lengths at {} CPU: {} GPU: {}".format(path, len(cpu), len(gpu))
+        assert len(cpu) == len(gpu), "{} and GPU row have different lengths at {} CPU: {} GPU: {}".format(cpu_data_adapted_from, path, len(cpu), len(gpu))
```

Comment:
Do we want to change the second `CPU:` to also be `{}:`?

Reply:
Done.